modules/administration-guide/pages/defining-the-list-of-images-to-pull.adoc

```diff
@@ -7,13 +7,24 @@
 [id="defining-the-list-of-images-to-pull"]
 = Defining the list of images to pull
 
-The {image-puller-name-short} can pre-pull most images, including scratch images such as `che-machine-exec`. However, images that mount volumes in the Dockerfile, such as `traefik`, are not supported for pre-pulling on OpenShift 3.11.
+You can speed up workspace startup by using {image-puller-name-short} to pre-pull workspace-related images such as `che-code`, `che-machine-exec`, and the Universal Developer Image (`udi`) image.
+
+.Prerequisites
+
+* The `podman` CLI tool.
+* The `yq` CLI tool.
 
 .Procedure
 
-. Gather a list of relevant container images to pull by navigating to the `pass:c,a,q[{prod-url}]/plugin-registry/v3/external_images.txt` URL.
+* Get the latest {devworkspace} and {prod-short} operator bundle images, and parse the related images from the `ClusterServiceVersion` YAML file in the `/manifests/` folder into a simple list of container images:
 
-. Determine images from the list for pre-pulling. For faster workspace startup times, consider pulling workspace related images such as `che-theia`, `che-machine-exec`, `che-theia-endpoint-runtime-binary`, and plug-in sidecar images.
+[subs="+attributes"]
+----
+$ cd /tmp && \
+curl -sSLkO https://raw.githubusercontent.com/redhat-developer/devspaces/devspaces-3-rhel-8/product/getRelatedImages.sh && \
+chmod +x getRelatedImages.sh && \
+./getRelatedImages.sh -t {prod-ver-patch} -d {devworkspace-operator-version} --quiet
+----
 
```

Comment:
The script does not seem to work on OS X:

```
./getRelatedImages.sh -t 3.2 -d 0.16


[INFO] Using local quay.io/devspaces/devspaces-operator-bundle:3.2 (a7b1881d3b13)...
Error: open /dev/stdout: permission denied
find: /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.VJs2Fa9N/quay.io-*-devspaces-operator-bundle-3.2-*/manifests/: No such file or directory
[INFO] Checking CSV:
Error: unknown shorthand flag: 'r' in -r
Usage:
  yq [flags]
  yq [command]
```

I have my doubts about replacing incomplete solution with a broken one. Can this be reconsidered?

Reply:
which `yq` do you have installed? If it's the mikefaraq one, not the jq wrapper written in python, then you'll have a bad time. I can fluff up the script so it only works with the python based yq...

Reply:
Hi @nickboldt I reinstalled the yq with the python version but it still fails because of 
```
Error: open /dev/stdout: permission denied
```

Which I assume it's related to 
https://github.com/containers/podman/issues/12402
I am using Podman 4.3

It now fails with:
```
bash -x ./getRelatedImages.sh -t 3.2 -d 0.16
+ QUIET=
+ [[ 4 -gt 0 ]]
+ case $1 in
+ VERSION=3.2
+ shift 1
+ shift 1
+ [[ 2 -gt 0 ]]
+ case $1 in
+ DWO_VERSION=0.16
+ shift 1
+ shift 1
+ [[ 0 -gt 0 ]]
+ [[ ! -n 3.2 ]]
+ [[ ! -n 0.16 ]]
++ mktemp -d
+ TMPDIR=/var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP
+ cd /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP
+ [[ ! -f /tmp/containerExtract.sh ]]
+ curl -sSLO https://raw.githubusercontent.com/redhat-developer/devspaces/devspaces-3-rhel-8/product/containerExtract.sh
+ chmod +x /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/containerExtract.sh
+ EXTRACT_CMD='/var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/containerExtract.sh  --tmpdir /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP --delete-before --delete-after'
+ [[ 3.2 == \7\.* ]]
+ [[ 3.2 == \8\.* ]]
+ [[ 3.2 == \3\.* ]]
+ rm -fr '/var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/quay.io-*-devspaces-operator-bundle-3.2-*'
+ /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/containerExtract.sh --tmpdir /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP --delete-before --delete-after quay.io/devspaces/devspaces-operator-bundle:3.2
[INFO] Using local quay.io/devspaces/devspaces-operator-bundle:3.2 (a7b1881d3b13)...
Error: open /dev/stdout: permission denied
++ find '/var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/quay.io-*-devspaces-operator-bundle-3.2-*/manifests/' -name '*csv.yaml'
find: /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/quay.io-*-devspaces-operator-bundle-3.2-*/manifests/: No such file or directory
+ CSV=
+ [[ '' == '' ]]
+ echo '[INFO] Checking CSV: '
[INFO] Checking CSV:
+ yq -r '.spec.relatedImages[].image' ''
+ sort -uV
usage: yq [-h] [--yaml-output] [--yaml-roundtrip]
          [--yaml-output-grammar-version {1.1,1.2}] [--width WIDTH]
          [--indentless-lists] [--in-place] [--version]
          [jq_filter] [files ...]
yq: error: argument files: can't open '': [Errno 2] No such file or directory: ''
+ [[ '' == '' ]]
+ echo

+ rm -fr '/var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/quay.io-*-devworkspace-operator-bundle-0.16'
+ /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/containerExtract.sh --tmpdir /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP --delete-before --delete-after quay.io/devworkspace/devworkspace-operator-bundle:0.16
[INFO] Using local quay.io/devworkspace/devworkspace-operator-bundle:0.16 (0cf26f525cb1)...
Error: open /dev/stdout: permission denied
++ find '/var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/quay.io-*-devworkspace-operator-bundle-0.16-*/manifests/' -name '*clusterserviceversion.yaml'
find: /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/quay.io-*-devworkspace-operator-bundle-0.16-*/manifests/: No such file or directory
+ CSV=
+ [[ '' == '' ]]
+ echo '[INFO] Checking CSV: '
[INFO] Checking CSV:
+ yq -r '.spec.relatedImages[].image' ''
+ sort -uV
usage: yq [-h] [--yaml-output] [--yaml-roundtrip]
          [--yaml-output-grammar-version {1.1,1.2}] [--width WIDTH]
          [--indentless-lists] [--in-place] [--version]
          [jq_filter] [files ...]
yq: error: argument files: can't open '': [Errno 2] No such file or directory: ''
+ [[ '' == '' ]]
+ echo

+ cd /tmp
+ rm -fr /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP
```

If the script already requires docker/podman I believe containerizing would make most sense as it completely removes the risk of running into env specific issue.

It would be also nice if it supported configuring container engine, i.e. ./script .sh-ce DOCKER

Reply:
I see you also mentioned `chectl`, that would be even better alternative. I still think this should be more of a supplementary solution and the real one should be improving the _default_ image list when image puller is enabled. For many use cases the default list would a great starting point already, so I'd prefer not to manually configuring it - especially if I will just reuse output of `./getRelatedImages.sh`.

Reply:
re: docker, um, don't you work for Red Hat? :P Eat the dogfood, make it better. :) 

re: podman, I use 4.3.0 too, but on fedora 36. I've also set it up to be rootless. https://github.com/containers/podman/blob/main/docs/tutorials/rootless_tutorial.md

you could also debug the containerExtract.sh script like this:

```
bash -x /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP/containerExtract.sh --tmpdir /var/folders/yh/dtn4z9nd0cgf_y0trh7znd6m0000gn/T/tmp.7QvlykqP --delete-before --delete-after quay.io/devspaces/devspaces-operator-bundle:3.2
```

Reply:
agreed -- here's another approach that @amisevsk wrote yesterday... haven't tested it yet. https://gist.github.com/amisevsk/9df020a46156b1824cd71a45785449aa

Reply:
It would be interesting to combine these approaches. i.e. run your bash script inside the container as shown above. It  wouldn't need serviceAccount/RoleBindings, so the YAML would be quite minimal.  And it would give you a chance to review (==and pick&choose) those images before applying those, as the approach above directly sets those, rather silently. And it also removes any environmental factor, as the script is now running into a fixed environment all the time.